### ISSUE TITLE
translate-c generates zig code with unnecessary `_ = foo;` statements for unused local variables and parameters

### ISSUE THREAD

**Issue by andrewrk** [ziglang/zig#9205]
```c
void foo(void) {
    int x;
}
```

This code translates to:

```zig
pub export fn foo() void {
    var x: c_int = undefined;
}
```

If you try to compile it with self-hosted:

```
$ ./zig build-obj test.zig
test.zig:2:9: error: unused local variable
    var x: c_int = undefined;
        ^
```

Instead it should generate:

```zig
pub export fn foo() void {
    var x: c_int = undefined;
    _ = x;
}
```

However ideally it would keep track of which variables and parameters are used and not generate unnecessary `_ = foo;` statements.

This is blocking #9191.

**Comment by ehaas** [comment]
I think the `_ = x;` solution would be pretty quick to implement as an unblocker, then maybe we could circle back and make it smarter about not emitting those for used variables?

**Comment by andrewrk** [comment]
Sounds good to me!

**Comment by ehaas** [comment]
Cool, can have that to you later tonight

**Comment by ehaas** [comment]
@andrewrk mostly done; however `__USER_LABEL_PREFIX__` is causing a problem on this branch. It gets injected automatically by clang https://gcc.gnu.org/onlinedocs/cpp/Common-Predefined-Macros.html

It translates to: `pub const __USER_LABEL_PREFIX__ = _;`

Which causes

```
test.zig:273:35: error: '_' may not be used as an identifier
pub const __USER_LABEL_PREFIX__ = _;
                                  ^
```

Even when the symbol is not otherwise referred to.

Seems like I could either special case that particular macro (skip it) or look for ones that have a single `_` identifier as the definition.

**Comment by ehaas** [comment]
Actually, one other issue which is the removal of packed/extern enums. I have a PR for that already - https://github.com/ziglang/zig/pull/9164 - should I bring that in as well?

**Comment by ehaas** [comment]
OK, PR for this one here: https://github.com/ziglang/zig/pull/9209

Assuming it looks good, I can rebase #9164 onto `stage1-astcheck` after this one gets merged.

**Comment by andrewrk** [comment]
> Seems like I could either special case that particular macro (skip it) or look for ones that have a single `_` identifier as the definition.

The intended solution for this is to use `@"_"` to refer to an identifier named `_`. Let me know if you hit any bugs related to that.

**Comment by ehaas** [comment]
> The intended solution for this is to use @"_" to refer to an identifier named _. Let me know if you hit any bugs related to that.

Is that expected to work now? I get the same error for `pub const FOO = @"_";`

```
test.zig:1:17: error: '_' may not be used as an identifier
pub const FOO = @"_";
                ^
```

**Comment by andrewrk** [comment]
> Is that expected to work now? I get the same error for `pub const FOO = @"_";`

Oops, forgot to answer this question. I believe I fixed it in abfee127353cbdadab4282b61056222cdfbec2d4.

**Comment by ehaas** [comment]
> > Is that expected to work now? I get the same error for `pub const FOO = @"_";`
> 
> Oops, forgot to answer this question. I believe I fixed it in [abfee12](https://github.com/ziglang/zig/commit/abfee127353cbdadab4282b61056222cdfbec2d4).

I'm able to declare a public symbol with that commit (eg. `pub const @"_": c_int = 5;` works) however I'm not able to declare a variable: 

```zig
pub export fn main() void {
    var @"_": c_int = 5;
    _ = @"_";
}
```


```
./test.zig:2:5: error: `_` is not a declarable symbol
    var @"_": c_int = 5;
    ^
./test.zig:3:9: error: `_` may only be used to assign things to
    _ = @"_";
        ^
```
